src/vs/workbench/contrib/debug/browser/debugService.ts

```diff
@@ -566,6 +569,11 @@ export class DebugService implements IDebugService {
 				dataBreakpoints.forEach(dbp => this.model.removeDataBreakpoints(dbp.getId()));
 			}
 
+			const closeConsoleOnEnd = this.configurationService.getValue<IDebugConfiguration>('debug').closeConsoleOnEnd;
+			if (closeConsoleOnEnd === 'always' || (closeConsoleOnEnd === 'whenOpenedByDebug' && !this.internalTerminalWasOpened)) {
```

Comment:
You shouold check if the activePanel is the debug console, you can not just close any panel. Since the user might have opened a terminal for example